cms/djangoapps/contentstore/middleware.py

```diff
@@ -0,0 +1,77 @@
+"""
+Middlewares for contentstore app.
+"""
+
+import logging
+
+from django.core.exceptions import PermissionDenied
+from django.urls import Resolver404, resolve
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import AssetKey, CourseKey, UsageKey
+
+from common.djangoapps.student.roles import CourseLimitedStaffRole
+
+logger = logging.getLogger(__name__)
+
+
+class LimitedStaffAccessMiddleware:
```

Comment:
Maybe `StudioPermissionsMiddleware` instead, to more clearly convey its purpose? It's not to just check for Limited Staff Access users, but to check for any user that should not have Studio access for a given course.